report/metrics.go

```diff
@@ -52,28 +53,55 @@ type Sample struct {
 	Value     float64   `json:"value"`
 }
 
-var nilMetric = Metric{Samples: ps.NewList()}
+// MakeSingletonMetric makes a metric with a single value
+func MakeSingletonMetric(t time.Time, v float64) Metric {
```

Comment:
> Could possibly do `func MakeMetric(samples ...Sample) Metric`

That's what I did originally. Problem is ... `go vet` doesn't like `Sample{t, v}`:

```
report.Sample composite literal uses unkeyed fields
```

it wants `report.Sample{Timestamp: t, Value: v}` which is incredibly long and verbose. `[]report.Sample{{t1,v1},{t2,v2}}` is perfectly fine, which is what I did, also adding the `MakeSingletonMetric()` function because we do this very often.